cloud/libvirt/actuators/machine/utils/volume.go

```diff
@@ -171,6 +158,17 @@ func CreateVolume(volumeName, poolName, baseVolumeID, source, volumeFormat strin
 		}
 
 		// create the volume
+		pool, err := virConn.LookupStoragePoolByName(poolName)
+		defer pool.Free()
+
+		// Refresh the pool of the volume so that libvirt knows it is
+		// not longer in use.
+		waitForSuccess("error refreshing pool for volume", func() error {
+			return pool.Refresh(0)
```

Comment:
How does this exit? Will it run forever if the pool fails to refresh successfully?